### ISSUE TITLE
Sniffing from STDIN shows Generic.Files.LowercasedFilename.NotFound error

### ISSUE THREAD

**Issue by skyrpex** [squizlabs/PHP_CodeSniffer#711]
Atom's PHPCS linter plugin uses now STDIN to lint on the fly. Someone that uses the [Wordpress-VIP ruleset](https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/blob/develop/WordPress-VIP/ruleset.xml) is getting this error Generic.Files.LowercasedFilename.NotFound.

The output of this command ```cat "path/myfile.php" | phpcs --report=json --standard=WordPress-VIP``` is:

```json
{
    "totals": {
        "errors": 1,
        "warnings": 0,
        "fixable": 0
    },
    "files": {
        "STDIN": {
            "errors": 1,
            "warnings": 0,
            "messages": [
                {
                    "message": "Filename \"STDIN\" doesn't match the expected filename \"stdin\"",
                    "source": "Generic.Files.LowercasedFilename.NotFound",
                    "severity": 5,
                    "type": "ERROR",
                    "line": 1,
                    "column": 1,
                    "fixable": false
                }
            ]
        }
    }
}
```

You'll notice that STDIN is treated the filename... See AtomLinter/linter-phpcs#60

**Comment by gsherwood** [comment]
The proper way to resolve this is for the plugin to tell PHPCS the filename so that sniffs that check the filename still work correctly. To do that, the first line of the input should be `phpcs_input_file: [file path]`, for example:

```php
phpcs_input_file: /path/to/test.php
<?php
echo 'test';

**Comment by gsherwood** [comment]
I've posted the same comment on the linked issue for the plugin. This feature was added in version 2.0 specifically to allow for plugin (Sublime plugins, at the time) to support filename checking.

**Comment by gsherwood** [comment]
Now that I've said that, the quick and easy way to fix this is for the sniff to ignore files with the name STDIN, which I'll also do. But that's something that will need to wait for the next release while the Atom plugin could be updated to work correctly today.
